service-runner.js

```diff
@@ -143,6 +146,24 @@ ServiceRunner.prototype.updateConfig = function updateConfig(conf) {
     }
 };
 
+ServiceRunner.prototype._checkHeartbeat = function() {
+    var self = this;
+    self.interval = setInterval(function() {
+        if (!self._shuttingDown && !self._inRollingRestart) {
+            Object.keys(cluster.workers).forEach(function(workerId) {
+                var worker = cluster.workers[workerId];
+                var lastBeat = self.workerHeartbeatTime[worker.process.pid];
+                if (!lastBeat || (!lastBeat.killed && new Date() - lastBeat.time > self.config.timeout)) {
+                    self._logger.log('error/service-runner/master',
+                        'worker ' + worker.process.pid + ' stopped sending heartbeats, killing.');
+                    self._stopWorker(workerId);
+                    // Don't need to respawn a worker, it will be restarted upon 'exit' event
+                }
+            });
+        }
+    }, self.config.timeout);
```

Comment:
I'm wondering if we should set an interval shorter than the time-out in order not to miss some workers that should be killed; e.g. in cases where this function runs slightly before a worker's time runs out, we'd have to wait twice as much to kill it. So, perhaps a skewed interval would be better here, such as `self.config.timeout / 2 + 1` (or even `/ 3 + 1`).